### ISSUE TITLE
`npm run lint` deadlocks

### ISSUE THREAD

**Issue by mwu-tow** [enso-org/enso#8898]
Sometimes CI Lint step fails.

`npm run lint` is invoked (on the repo-root level). This command never finishes. The output is:
```

> lint
> npm --workspaces --if-present run lint


> enso-ide-desktop@1.0.0 lint
> npm run --workspace=enso-gui2 compile-server && npm run lint-only


> enso-gui2@0.1.0 compile-server
> tsc -p tsconfig.server.json


> enso-ide-desktop@1.0.0 lint-only
> eslint .


> enso-gui2@0.1.0 lint
> eslint .
```

This has happened at least twice on CI:
* https://github.com/enso-org/enso/actions/runs/7449914516/job/20267638401?pr=8707
* https://github.com/enso-org/enso/actions/runs/7699140312/job/20980061367

Both times with the same output, suggesting that it is the `enso-gui2` linting specifically that causes deadlock.

**Comment by Akirathan** [comment]
Another instance of this problem here https://github.com/enso-org/enso/actions/runs/7743955599/job/21116570129?pr=8764

<blockquote><img src="https://opengraph.githubassets.com/2a9641ed4b1466bd19a928c94a904a7a4cca95e0d9f5a8c6496ff75a3eccdd63/enso-org/enso" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/enso-org/enso">Fix Truffle invariants for Nothing with a warning · enso-org/enso@ad5ec40</a></strong></div><div>Hybrid visual and textual functional programming. Contribute to enso-org/enso development by creating an account on GitHub.</div></blockquote>

**Comment by farmaazon** [comment]
Seems to be fixed by #8947 as I don't see those errors since a week.
